src/main/java/uk/gov/hmcts/opal/interceptor/AcceptHeaderInterceptor.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.opal.interceptor;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.method.HandlerMethod;
+import uk.gov.hmcts.opal.annotation.CheckAcceptHeader;
+
+public class AcceptHeaderInterceptor implements HandlerInterceptor {
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
+        throws Exception {
+        if (handler instanceof HandlerMethod) {
+            HandlerMethod handlerMethod = (HandlerMethod) handler;
```

Comment:
You don't need to do the cast on it's own line.
See Pattern Matching 
https://blogs.oracle.com/javamagazine/post/pattern-matching-for-instanceof-in-java-14

Reply:
thanks for the tip, I've updated that line